src/EditorFeatures/CSharpTest/CodeActions/InvertIf/InvertIfTests.cs

```diff
@@ -0,0 +1,715 @@
+        public async Task TestSimplifyToLengthEqualsZero2()
+        {
+            await TestFixOneAsync(
+@"string[] x; [||]if (x.Length > 0) { GreaterThanZero(); } else { EqualsZero(); } } } ",
+@"string[] x; if (x.Length == 0) { EqualsZero(); } else { GreaterThanZero(); } } } ");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)]
+        public async Task TestSimplifyToLengthEqualsZero3()
+        {
+            await TestFixOneAsync(
+@"string x; [||]if (x.Length > 0x0) { a(); } else { b(); } } } ",
+@"string x; if (x.Length == 0x0) { b(); } else { a(); } } } ");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)]
+        public async Task TestSimplifyToLengthEqualsZero4()
+        {
+            await TestFixOneAsync(
+@"string x; [||]if (0 < x.Length) { a(); } else { b(); } } } ",
+@"string x; if (0 == x.Length) { b(); } else { a(); } } } ");
+        }
+
+        [WorkItem(545986, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/545986")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)]
+        public async Task TestSimplifyToLengthEqualsZero5()
+        {
+            await TestFixOneAsync(
+@"byte x = 1; [||]if (0 < x) { a(); } else { b(); } } } ",
+@"byte x = 1; if (0 == x) { b(); } else { a(); } } } ");
+        }
+
+        [WorkItem(545986, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/545986")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)]
+        public async Task TestSimplifyToLengthEqualsZero6()
+        {
+            await TestFixOneAsync(
+@"ushort x = 1; [||]if (0 < x) { a(); } else { b(); } } } ",
+@"ushort x = 1; if (0 == x) { b(); } else { a(); } } } ");
+        }
+
+        [WorkItem(545986, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/545986")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)]
+        public async Task TestSimplifyToLengthEqualsZero7()
+        {
+            await TestFixOneAsync(
+@"uint x = 1; [||]if (0 < x) { a(); } else { b(); } } } ",
+@"uint x = 1; if (0 == x) { b(); } else { a(); } } } ");
+        }
+
+        [WorkItem(545986, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/545986")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)]
+        public async Task TestSimplifyToLengthEqualsZero8()
+        {
+            await TestFixOneAsync(
+@"ulong x = 1; [||]if (0 < x) { a(); } else { b(); } } } ",
+@"ulong x = 1; if (0 == x) { b(); } else { a(); } } } ");
+        }
+
+        [WorkItem(545986, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/545986")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)]
+        public async Task TestSimplifyToLengthEqualsZero9()
+        {
+            await TestFixOneAsync(
+@"ulong x = 1; [||]if (0 == x) { a(); } else { b(); } } } ",
+@"ulong x = 1; if (0 < x) { b(); } else { a(); } } } ");
```

Comment:
i do not like that this flipped from ```==``` to  ```<```.  I think flipping hte other direction is fine.  But i do not want my strict equality relationals becoming comparisons.